WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -484,6 +484,15 @@ def update
         redirect_to root_url
       end
     elsif @competition.update_attributes(comp_params_minus_id)
+      if new_id && !@competition.update_attributes(id: new_id)
+        # Changing the competition id breaks all our associations, and our view
+        # code was not written to handle this. Rather than trying to update our view
+        # code, just revert the attempted id change. The user will have to deal with
+        # editing the ID text box manually. This will go away once we have proper
+        # immutable ids for competitions.
+        @competition = Competition.find(params[:id])
+      end
+
```

Comment:
The mail sending is an asynchronous job, so we want to pass to it the updated competition object (including id update). That's why I moved this id updating logic above mailer calls.